packages/web/src/pages/chat-page/components/ChatMessageList.tsx

```diff
@@ -54,18 +55,30 @@ const isScrolledToTop = (element: HTMLElement) => {
 }
 
 /**
- * Checks if the current message is the first unread message in the given chat
- * Used to render the new messages separator indicator
+ * Checks if the current message:
+ * - Is the first unread message
+ * - Is by a different user than the current one
+ * - Is at the index that matches the unread count (ie there's been no new messages since)
  */
-const isFirstUnread = (
-  chat: UserChat,
-  message: ChatMessage,
-  currentUserId: string | null,
-  prevMessage?: ChatMessage
-) =>
-  message.created_at > chat.last_read_at &&
-  (!prevMessage || prevMessage.created_at <= chat.last_read_at) &&
-  message.sender_user_id !== currentUserId
+const shouldRenderUnreadIndicator = (
+  unreadCount: number,
+  lastReadAt: string | undefined,
```

Comment:
optional `?` instead of `| undefined`?

Reply:
Unf can't in this context as it's not an object being passed, it's individual args. If the `lastReadAt` was optional, the args following it would need to be as well